docs/js-routes.md

```diff
@@ -0,0 +1,60 @@
+# Client-side routes
+
+_This document description how to use routes in the client-side JavaScript code._
+
+## Usage
+
+In the top-level React component, you need to import `subset` function from `utils/routes` module and call that function with the named routes you want to use. After that, you need to pass the newly created route subset object to the children, if they need routes. Note that you can pass default options to the `subset` function.
+
+```js
+// App.js
+import { subset } from '../utils/routes';
+
+export default (props, railsContext) => {
+  const routes = subset([
+    'homepage',
+  ], { locale: "en" })
+
+  return r(MyComponent, { routes });
+};
+```
+
+In the component, you then use the routes subset to create the routes you want:
+
+// MyComponent.js
+
+class MyComponent extends React.Component {
+  render() {
+    return a(href: this.props.root.homepage_path())
+  }
+}
+
+
+## Parameters
+
+You can give parameters to the route helper function as you would do in Rails:
```

Comment:
Minor thing about naming, parameters vs arguments. Parameters are the things you give names in function signature and arguments are the values you pass in when you call a function. So this section is about arguments, not about parameters.

ref: http://stackoverflow.com/questions/1788923/parameter-vs-argument